src/code/middleware/state-save.js

```diff
@@ -41,21 +41,43 @@ export default () => store => next => action => {
 
     // Store updated gems if they have changed
     if (action.type !== actionTypes.LOAD_SAVED_STATE &&
-          (JSON.stringify(prevState.gems) !== JSON.stringify(nextState.gems) ||
-           JSON.stringify(prevState.remediationHistory) !== JSON.stringify(nextState.remediationHistory))) {
-      let {gems, remediationHistory} = nextState;
+      (JSON.stringify(prevState.gems) !== JSON.stringify(nextState.gems)
+      // TODO: Uncomment once remediation is fixed!
+        // ||
+        // nextState.remediationHistory && JSON.stringify(prevState.remediationHistory) !== JSON.stringify(nextState.remediationHistory)
+      )) {
+      let {
+        gems
+        // , remediationHistory
+      } = nextState;
       userDataUpdate.state = {
         gems,
-        remediationHistory,
+        // remediationHistory,
         stateVersion: currentStateVersion
       };
     }
-    firebase.database().ref(userQueryString).update(userDataUpdate);
+    firebase.database().ref(userQueryString).update(userDataUpdate, (error, userDataUpdate) => {
+      if (error) {
+        console.logError("Error updating user state!", userDataUpdate, error);
+        return error;
```

Comment:
I'm not seeing any code that's dealing with this error return value. If the point is just to log the error, then maybe the returns can be removed for simplification.

Reply:
I want to include more thorough understanding of firebase connection state issues and handle failed updates more gracefully. The above code didn't even get hit when the Firebase update failed (due to undefined data being passed in the remediation section) - definitely flagging this in PT for further investigation.